src/app/signup/page.tsx

```diff
@@ -83,18 +129,20 @@ const SignUp = () => {
           </div>
           <div className="flex items-center px-12 mb-10 w-full">
             <form
-              // @ts-expect-error Type mismatch
-              onSubmit={handleSubmit((data) => signupUser(data))}
+              ref={formRef}
+              action={formAction}
+              onSubmit={handleSubmit(() => formRef.current?.submit())}
```

Comment:
Ah! So that's how you do it with hook form & server actions lol Thank you for fixing this!

Reply:
Yeah, the resources you gave were helpful. Thanks for that